File: py-appscript/trunk/sample/appscript/AB_list_people_with_emails.py
#!/usr/bin/env python

# Lists the name and email(s) of every person in Address Book
# with one or more email addresses.

from appscript import *

peopleref = app('Address Book').people[its.emails != []]
for name, emails in zip(peopleref.name.get(), peopleref.emails.value.get()):
	print name, emails

File: rb-appscript/tags/rb-appscript-0.6.0/sample/AB_list_people_with_emails.rb
#!/usr/bin/env ruby

# Lists the name and email(s) of every person in Address Book with
# one or more email addresses.

# Note: if using the appscript gem, rubygems must be required first:
begin; require 'rubygems'; rescue LoadError; end

require "appscript"
include Appscript

people_ref = app('Address Book').people[its.emails.ne([])]
p people_ref.name.get.zip(people_ref.emails.value.get)